File: python/p004.py
# Find largest palindrome that is a multiple of 2 3 digit numbers
from utils import is_palindrome


def compute():
    max_p = 0
    for i in range(100, 1000):
        # Only go to i (inclusive) to eliminate double counting
        for j in range(100, i + 1):
            num = i * j
            if num > max_p and is_palindrome(num):
                max_p = num
    return max_p


if __name__ == '__main__':
    print(compute())

File: c/p004.c
#include <stdio.h>
#include "utils.h"

int main()
{
    long num;
    long max_p = 0;
    for(int i = 100; i < 1000; i++)
    {
        for(int j = 100; j < i + 1; j++)
        {
            num = i * j;
            if (is_palindrome(num) && num > max_p) 
            {
                max_p = num;
            }
        }
    }
    printf("%ld\n", max_p);
}